src/ui/state/stakeholder.rs

```diff
@@ -146,3 +192,171 @@ impl From<StakeholderNetworkState> for Box<dyn State> {
         Box::new(s)
     }
 }
+
+#[derive(Debug)]
+pub struct StakeholderACKFundsState {
+    revaultd: Arc<RevaultD>,
+    warning: Watch<Error>,
+
+    balance: u64,
+    deposits: Vec<Deposit>,
+    view: StakeholderACKFundsView,
+}
+
+impl StakeholderACKFundsState {
+    pub fn new(revaultd: Arc<RevaultD>) -> Self {
+        StakeholderACKFundsState {
+            revaultd,
+            warning: Watch::None,
+            deposits: Vec::new(),
+            view: StakeholderACKFundsView::new(),
+            balance: 0,
+        }
+    }
+
+    fn update_deposits(&mut self, vaults: Vec<Vault>) -> Command<Message> {
+        self.calculate_balance(&vaults);
+        self.deposits = vaults.into_iter().map(|vlt| Deposit::new(vlt)).collect();
+        if let Some(Deposit::Pending { vault }) = self.deposits.first() {
+            return Command::perform(
+                get_revocation_txs(self.revaultd.clone(), vault.outpoint()),
+                |res| Message::Deposit(0, DepositMessage::RevocationTransactions(res)),
+            );
+        }
+        Command::none()
```

Comment:
nit: untested but i wonder if you could avoid allocating an entire vector of deposits to just use the first one (ie calling `first()` on `vaults`)

Reply:
every vaults are displayed, only the first has the revocation transactions loaded because we start by it to presign transactions, a dropdown is displayed with the signer